can/interfaces/slcan.py

```diff
@@ -252,10 +253,10 @@ def shutdown(self) -> None:
         self.serialPortOrig.close()
 
     def fileno(self) -> int:
-        if hasattr(self.serialPortOrig, "fileno"):
+        try:
             return self.serialPortOrig.fileno()
-        # Return an invalid file descriptor on Windows
-        return -1
+        except io.UnsupportedOperation:
+            raise NotImplementedError("fileno is not implemented using current CAN bus on this platform")
```

Comment:
Fix `black` formatter.

```suggestion
            raise NotImplementedError(
                "fileno is not implemented using current CAN bus on this platform"
            )
```